cloud/shared/bin/lib/terraform.sh

```diff
@@ -18,24 +13,22 @@ readonly TERRAFORM_APPLY="${TERRAFORM_BASE_COMMAND} \
 #######################################
 function terraform::perform_apply() {
   if [[ "${CIVIFORM_MODE}" == "dev" ]]; then
-    "$(${TERRAFORM_BASE_COMMAND} init -upgrade)"
+    terraform "${TERRAFORM_DIR}" init -upgrade
   else
     "cloud/${CIVIFORM_CLOUD_PROVIDER}/bin/setup_tf_shared_state" \
       "${TERRAFORM_TEMPLATE_DIR}/${BACKEND_VARS_FILENAME}"
 
-    "$(
-      ${TERRAFORM_BASE_COMMAND}
+    terraform "${TERRAFORM_DIR}" \
       init \
-        -input=false \
-        -upgrade \
-        -backend-config="${BACKEND_VARS_FILENAME}"
-    )"
+      -input=false \
+      -upgrade \
+      -backend-config="${BACKEND_VARS_FILENAME}"
   fi
 
   if azure::is_service_principal; then
-    "$(${TERRAFORM_APPLY} -auto-approve)"
+    "${TERRAFORM_APPLY[@]}" -auto-approve
```

Comment:
just for my understanding: [@] here is for applying the arguments correctly?

Reply:
This is how you expand the elements of the array: https://www.baeldung.com/linux/store-command-in-variable